src/charm.py

```diff
@@ -145,32 +137,32 @@ def _configure_nrf(self, event: EventBase) -> None:
             return
         if not self._get_database_uri():
             self.unit.status = WaitingStatus("Waiting for database URI")
-            event.defer()
             return
         if not self._container.exists(path=BASE_CONFIG_PATH):
             self.unit.status = WaitingStatus("Waiting for storage to be attached")
-            event.defer()
             return
         if not _get_pod_ip():
             self.unit.status = WaitingStatus("Waiting for pod IP address to be available")
-            event.defer()
             return
-        if not self._certificate_is_stored():
+        if not self._private_key_is_stored():
+            self._generate_private_key()
+        if not self._csr_is_stored():
+            self._request_new_certificate()
+        certificate_changed = False
```

Comment:
We can remove line 151.  The reason:
if provider_certificate exists, `certificate_changed` variable will be defined in line 154.
If provider certificate does not exist, return will exit the method and line 162 does not run unless line 153 returns True.

Reply:
That just feels wrong to me to not define a variable in the scope that it is going to be used.  Anyone else have thoughts on this?

Reply:
I also feel uncomfortable if we do not define the variable.

Reply:
Avoided the issue with a refactor